File: frontend/types/pesquisa.ts
export type PesquisaUsuario = {
    id: number;
    nome: string;
    nick: string;
    image?: string;
};

export type PesquisaLivro = {
    id?: number;
    titulo: string;
    autor: string;
    capa_url?: string;
    google_volume_id?: string;
    isbn?: string;
};

export type PesquisaResultado = {
    usuarios: PesquisaUsuario[];
    livros: PesquisaLivro[];
};

export function hrefLivroPesquisa(livro: PesquisaLivro): string {
    if (livro.id) return `/livros/${livro.id}`;
    if (livro.google_volume_id) return `/livros/pesquisa/${livro.google_volume_id}`;
    return "#";
}

export function keyLivroPesquisa(livro: PesquisaLivro, index: number): string {
    return String(livro.id || livro.google_volume_id || index);
}

File: backend/src/modelos/Pesquisa.go
package modelos

type PesquisaGlobalResponse struct {
	Usuarios []UsuarioPesquisa `json:"usuarios"`
	Livros   []LivroPesquisa   `json:"livros"`
}

type UsuarioPesquisa struct {
	ID    uint64 `json:"id"`
	Nome  string `json:"nome"`
	Nick  string `json:"nick"`
	Image string `json:"image,omitempty"`
}

type LivroPesquisa struct {
	ID             uint64 `json:"id"`
	Titulo         string `json:"titulo"`
	Autor          string `json:"autor"`
	CapaURL        string `json:"capa_url,omitempty"`
	GoogleVolumeID string `json:"google_volume_id,omitempty"`
	ISBN           string `json:"isbn,omitempty"`
}
